content/en/docs/tasks/tools/install-kubectl.md

```diff
@@ -295,63 +295,111 @@ kubectl cluster-info dump
 
 ## Enabling shell autocompletion
 
-kubectl includes autocompletion support, which can save a lot of typing!
+kubectl provides autocompletion support for Bash and Zsh, which can save you a lot of typing!
 
-The completion script itself is generated by kubectl, so you typically just need to invoke it from your profile.
+The procedure to properly set up autocompletion is different for Bash and Zsh. Both cases are explained below. 
 
-Common examples are provided here. For more details, consult `kubectl completion -h`.
+- [Bash](#bash)
+- [Zsh](#zsh)
 
-### On Linux, using bash
-On CentOS Linux, you may need to install the bash-completion package which is not installed by default.
+### Bash
 
-```shell
-yum install bash-completion -y
-```
+The kubectl completion script for Bash can be generated with `kubectl completion bash`. Sourcing this script in your shell enables kubectl autocompletion.
 
-To add kubectl autocompletion to your current shell, run `source <(kubectl completion bash)`.
+However, the completion script depends on [**bash-completion**](https://github.com/scop/bash-completion) (you can test if you already have bash-completion installed by running the command `type _init_completion`).
 
-To add kubectl autocompletion to your profile, so it is automatically loaded in future shells run:
+The below sections walk you through installing bash-completion and setting up kubectl autocompletion for both Linux and macOS.
 
-```shell
-echo "source <(kubectl completion bash)" >> ~/.bashrc
-```
+#### Bash on Linux
 
-### On macOS, using bash
-On macOS, you will need to install bash-completion support via [Homebrew](https://brew.sh/) first:
+bash-completion is available in many package managers (see [here](https://github.com/scop/bash-completion#installation)). You can install it, for example, with `apt-get install bash-completion` or `yum install bash-completion`.
 
-```shell
-## If running Bash 3.2 included with macOS
-brew install bash-completion
-## or, if running Bash 4.1+
+The above commands create `/usr/share/bash-completion/bash_completion`, which is the main script of bash-completion. Depending on your package manager, you have to manually source this file in your `~/.bashrc` file.
+
+To find out, restart your shell and run `type _init_completion`. If the command succeeds, you're already set, otherwise add the following to your `~/.bashrc` file:
+
+~~~bash
+source /usr/share/bash-completion/bash_completion
+~~~
+
+Restart your shell and verify that bash-completion is correctly installed with `type _init_completion`.
+
+You now need to take care of sourcing the kubectl completion script in your shell sessions. There are two ways in which you can do this:
+
+- Source the completion script in your `~/.bashrc` file:
+
+    ~~~bash
```

Comment:
nit: apart from the fact you need to use the backticks, also please remove the space before the backticks, it doesn't add any benefit however doing so we keep the consistency

Reply:
I indented the code blocks to not break the bulleted list (code blocks are indented the same amount as the list text in the rendered view). But of course it's a matter of taste, the code blocks might also start from the left edge like content that is not in a list.

Reply:
okay